test/transcoding/SPV_INTEL_fpga_dsp_control/prefer_dsp.ll

```diff
@@ -0,0 +1,84 @@
+; template<typename Function>
+; [[intel::prefer_dsp]]
+; void math_prefer_dsp_propagate(Function f)
+; {
+;   f();
+; }
+
+; int main() {
+;   math_prefer_dsp_propagate([]() {
+;     int a = 0;
+;     a += 1;
+;   });
+
+; return 0;
+; }
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_fpga_dsp_control -o %t.spv
+; RUN: llvm-spirv %t.spv --to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: 2 Capability MathOpDSPModeINTEL
+; CHECK-SPIRV: 8 Extension "SPV_INTEL_fpga_dsp_control"
```

Comment:
The number before instruction name might be changed if you change the test, I suggest not hard code it in check lines.